### ISSUE TITLE
ofGLProgrammableRenderer: can't change line width

### ISSUE THREAD

**Issue by ghost** [openframeworks/openFrameworks#3460]
Line example:

``` cpp
#include <ofMain.h>

class ofApp : public ofBaseApp
{
  public:
    void setup  ();
    void draw   ();
};

void ofApp::setup()
{
}

void ofApp::draw()
{
  ofSetLineWidth(20.0f);
  ofSetColor(ofColor::tomato);
  ofDrawLine(10, 10, ofGetWidth()-10, ofGetHeight()-10);
}

int main()
{
  ofSetOpenGLVersion(2,1);
  ofSetupOpenGL(280, 280, OF_WINDOW);
  ofRunApp(new ofApp());
}
```

OpenGL 2.1:

![](http://i.imgur.com/9HwUr8m.png)

OpenGL 3.3 (same code with `ofSetOpenGLVersion(3,3)`):

![](http://i.imgur.com/8oM1uuF.png)

Path example:

``` cpp
#include <ofMain.h>

class ofApp : public ofBaseApp
{
  public:
    void setup  ();
    void draw   ();

    ofPath path;
};

void ofApp::setup()
{
  int w = ofGetWidth();
  int h = ofGetHeight();

  path.setFilled(false);
  path.setStrokeWidth(10.0f);
  path.setColor(ofColor::tomato);

  path.moveTo(  30,  30);
  path.lineTo(  30,h-30);
  path.lineTo(w-30,h-30);
  path.lineTo(w-30,  30);
  path.close();
}

void ofApp::draw()
{
  path.draw();
}

int main()
{
  ofSetOpenGLVersion(2,1);
  ofSetupOpenGL(280, 280, OF_WINDOW);
  ofRunApp(new ofApp());
}
```

OpenGL 2.1:

![](http://i.imgur.com/vTy9LQH.png)

OpenGL 3.3 (same code with `ofSetOpenGLVersion(3,3)`):

![](http://i.imgur.com/1UbsxfV.png)

**Comment by procedural** [comment]
Found [this comment](https://github.com/openframeworks/openFrameworks/blob/c473f23/libs/openFrameworks/gl/ofGLProgrammableRenderer.cpp#L868-L871) in older commit :)

**Comment by bilderbuchi** [comment]
wondering why that is commented out, though? @arturoc @tgfrerer ?

**Comment by tgfrerer** [comment]
I put this in commented, because prints to the log are very expensive, and a log message might get emitted more than once per draw loop - slowing the renderer down terribly.

if we had a mechanism in openFrameworks to log a warning only on first occurance, that would be ideal.

**Comment by bilderbuchi** [comment]
OTOH, it only prints if you're doing something you shouldn't, right? so if your lines are not thickening if you use setLineWidth, slow performance shouldn't matter that much, should it?

**Comment by tgfrerer** [comment]
That, on the other hand, is very true. +1 for uncommenting it, then =)

**Comment by bilderbuchi** [comment]
consensus from #3896 is also that this warning should go back in, maybe with a slightly changed formulation.

**Comment by procedural** [comment]
I've wrote a post on forum on how to workaround this issue: http://forum.openframeworks.cc/t/ofsetlinewidth-doesnt-work-with-ofglprogrammablerenderer-type/13664/12?u=procedural

**Comment by NickHardeman** [comment]
Closed by #8019
